commanderbot_ext/invite/invite_store.py

```diff
@@ -0,0 +1,127 @@
+from datetime import datetime
+from typing import Iterable, Optional, Set
+from dataclasses import dataclass
+
+from commanderbot_lib.database.abc.versioned_file_database import (
+    DataMigration,
+    VersionedFileDatabase,
+)
+from commanderbot_lib.store.abc.versioned_cached_store import VersionedCachedStore
+from discord import Guild, Message
+
+from commanderbot_ext.invite import invite_migrations as migrations
+from commanderbot_ext.invite.invite_cache import (
+    InviteCache,
+    InviteEntry,
+    InviteGuildData,
+)
+from commanderbot_ext.invite.invite_options import InviteOptions
+
+
+class NotExistException(Exception):
+    pass
+
+
+class NotApplicableException(Exception):
+    pass
+
+
+class TagNameClashException(Exception):
+    pass
+
+
+@dataclass
+class InviteNameException(Exception):
+    existing: Optional[InviteEntry]
+
+
+class InviteStore(
+    VersionedCachedStore[InviteOptions, VersionedFileDatabase, InviteCache]
+):
+    # @implements CachedStore
+    async def _build_cache(self, data: dict) -> InviteCache:
+        return await InviteCache.deserialize(data)
+
+    # @implements CachedStore
+    async def serialize(self) -> dict:
+        return self._cache.serialize()
+
+    # @implements VersionedCachedStore
+    def _collect_migrations(
+        self,
+        database: VersionedFileDatabase,
+        actual_version: int,
+        expected_version: int,
+    ) -> Iterable[DataMigration]:
+        if actual_version < 1:
```

Comment:
Same as the other comment: you can probably empty this function completely via `yield from ()` or `if False: yield` etc. Since there haven't been any historical changes to the data format yet.